### PR TITLE
feat: Add types to chat events to make it easier to switch on

### PR DESCRIPTION
**Pull request by hofmeister**

### CODE REVIEW

File: src/renderer/components/storm/context/chatUserInteractionAtoms.ts [hunk 1]
```diff
@@ -5,99 +5,52 @@
...
+          type: 'connection';
+          id: string;
+      }
+    | {
+          type: 'api';
+          blockName: string;
+      }
+    | {
+          type: 'none';
+      };
```
Comment (andertun) [inline src/renderer/components/storm/context/chatUserInteractionAtoms.ts:26]: This approach removes the possibility to have two things hovered, when the user hovers a data model I thought we might want to highlight both the block and the DB resource, so in that case a `blockName` and a `resourceName` would be needed. What do you think?

Comment (hofmeister) [reply src/renderer/components/storm/context/chatUserInteractionAtoms.ts:26]: The DB resource belongs to the block so both would need to react visually. But we generate the blocks in a way where they only ever have 1 DB resource and 1 API resource - so we basically just need the block name

Comment (andertun) [reply src/renderer/components/storm/context/chatUserInteractionAtoms.ts:26]: Ok, yes we can add `blockName` to those events.

**Review by andertun — APPROVED** [review]

File: src/renderer/components/storm/context/chatUserInteractionAtoms.ts (hunk 1 above)
Comment (hofmeister) [reply src/renderer/components/storm/context/chatUserInteractionAtoms.ts:26]: I think I did 👍

**Pull request merged**